kernel/src/main/java/org/kframework/kast/KastFrontEnd.java

```diff
@@ -110,13 +105,24 @@ public int run() {
                     sort = def.programStartSymbol;
                 }
             }
-            options.module = options.module == null ? def.mainSyntaxModuleName() : options.module;
+            Module compiledMod;
+            if (options.module == null) {
+                options.module = def.mainSyntaxModuleName();
+                switch (options.input) {
+                case KORE:
+                    compiledMod = def.languageParsingModule();
+                    break;
+                default:
+                    compiledMod = def.kompiledDefinition.getModule(def.mainSyntaxModuleName()).get();
+                }
+            } else {
+                compiledMod = def.kompiledDefinition.getModule(options.module).get();
+            }
             Option<Module> maybeMod = def.programParsingModuleFor(options.module, kem);
             if (maybeMod.isEmpty()) {
                 throw KEMException.innerParserError("Module " + options.module + " not found. Specify a module with -m.");
             }
             Module mod = maybeMod.get();
-            Module compiledMod = def.kompiledDefinition.getModule(options.module).get();
 
             K parsed = kread.prettyRead(mod, sort, def, source, FileUtil.read(stringToParse), options.input);
 
```

Comment:
We need to change the code which comes after this. The next `if` block should only check if `options.expandMacros`, and we should get rid of the `--kore` option (in favor of using `--input kore`).

So the second `if` block after this line should be removed, and we should go straight to the `else` branch.

Reply:
@traiansf pointed out that the code following this block is dealing with _output_ and not _input_ options, so this comment is incorrect.